File: week1/3-Stack/stack.py
class Stack:

    def __init__(self):
        self.myArray = []

    def size(self):
        return len(self.myArray)

    def push(self, value):
        self.myArray.append(value)

    def pop(self):
        last = self.myArray[Stack.size(self)-1]
        self.myArray = self.myArray[:-1]
        return last

    def peek(self):
        return self.myArray[0]

q = Stack()

q.push(1)
q.push(2)
q.push(100)
print(q.pop())
print(q.pop())

File: week1/3-Stack/Stack.java
<<<<<<< HEAD
package com.company;

public class Stack
{
    Queue queue = new Queue();

    public void Push(int value)
    {
        queue.Push(value);
    }

    public int Pop()
    {
        Queue newQueue = new Queue();
        int queueSize = queue.Size();

        for(int i = 0; i < queueSize - 1; i++)
        {
            newQueue.Push(queue.Pop());
        }

        int popValue = queue.Pop();

        queue = newQueue;

        return popValue;
    }

    public int Peak()
    {
        Queue newQueue = new Queue();
        int peakedValue = 0;
        int queueSize = queue.Size();

        for(int i = 0; i < queueSize; i++)
        {
            if(i == queueSize - 1)
                peakedValue = queue.Peak();

            newQueue.Push(queue.Pop());
        }

        queue = newQueue;

        return peakedValue;
    }

    public int Size()
    {
        return queue.Size();
    }
=======
class Stack<T> {

  // Adds value to the end of the Stack.
  // Complexity: O(1)
  public void push(T value) {
    // ...
  }

  // Returns value from the end of the Stack and removes it.
  // Complexity: O(n)
  public T pop() {
    // ...
  }

  // Returns value from the end of the Stack without removing it.
  // Complexity: O(n)
  public T peek() {
    // ...
  }

  // Returns the number of elements in the Stack.
  // Complexity: O(1)
  public int size() {
    // ...
  }
>>>>>>> upstream/master
}
